src/equation_of_state/MOM_EOS_base_type.F90

```diff
@@ -0,0 +1,464 @@
+!> A generic type for equations of state
+module MOM_EOS_base_type
+
+! This file is part of MOM6. See LICENSE.md for the license.
+
+implicit none ; private
+
+public EOS_base
+
+!> The base class for implementations of the equation of state
+type, abstract :: EOS_base
+
+contains
+
+  ! The following functions/subroutines are deferred and must be provided specifically by each EOS
+
+  !> Deferred implementation of the in-situ density as an elemental function [R ~> kg m-3]
+  procedure(i_density_elem), deferred :: density_elem
+  !> Deferred implementation of the in-situ density anomaly as an elemental function [R ~> kg m-3]
```

Comment:
As I understand it, all of these these density functions are returning values in MKS units, not rescaled units.  If this is so, the units on this line should be `[kg m-3]`, not `[R ~> kg m-3]` which would indicate rescaling.  A similar issue applies to the description of the spec_vol routines, and to all of the analogous interfaces in the other equation of state modules.

Reply:
Yes, those indeed crept in when copying the interfaces from the wrong file. Fixed.